py-polars/tests/unit/operations/namespaces/test_binary.py

```diff
@@ -210,6 +211,64 @@ def test_reinterpret(
         assert_frame_equal(result, expected_df)
 
 
+@pytest.mark.parametrize(
+    ("dtype", "inner_type_size", "struct_type"),
+    [
+        (pl.Array(pl.Int8, 3), 1, "b"),
+        (pl.Array(pl.UInt8, 3), 1, "B"),
+        (pl.Array(pl.UInt8, (3, 4, 5)), 1, "B"),
+        (pl.Array(pl.Int16, 3), 2, "h"),
+        (pl.Array(pl.UInt16, 3), 2, "H"),
+        (pl.Array(pl.Int32, 3), 4, "i"),
+        (pl.Array(pl.UInt32, 3), 4, "I"),
+        (pl.Array(pl.Int64, 3), 8, "q"),
+        (pl.Array(pl.UInt64, 3), 8, "Q"),
+        (pl.Array(pl.Float32, 3), 4, "f"),
+        (pl.Array(pl.Float64, 3), 8, "d"),
+    ],
+)
+def test_reinterpret_list(
+    dtype: pl.Array,
+    inner_type_size: int,
+    struct_type: str,
+) -> None:
+    # Make test reproducible
+    random.seed(42)
+
+    type_size = inner_type_size
+    shape = dtype.shape
+    if isinstance(shape, int):
+        shape = (shape,)
+    for dim_size in dtype.shape:
+        type_size *= dim_size
+
+    byte_arr = [random.randbytes(type_size) for _ in range(3)]
```

Comment:
What happens if byte strings are different lengths?

Reply:
As far as _desirable_ behavior:

1. Pad with nulls (only applicable to too-short).
2. Replace with null.
3. Error out completely.

One imagines it might be desirable to have different behavior for too-short and too-long.

Reply:
This is blocked by the reshaping logic, which doesn't do what we want with nulls.